pkg/scheduler/framework/plugins/dynamicresources/dynamicresources_test.go

```diff
@@ -0,0 +1,660 @@
+limitations under the License.
+*/
+
+package dynamicresources
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sort"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	v1 "k8s.io/api/core/v1"
+	resourcev1alpha1 "k8s.io/api/resource/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	corev1ac "k8s.io/client-go/applyconfigurations/core/v1"
+	metav1ac "k8s.io/client-go/applyconfigurations/meta/v1"
+	resourcev1alpha1ac "k8s.io/client-go/applyconfigurations/resource/v1alpha1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes/fake"
+	cgotesting "k8s.io/client-go/testing"
+	"k8s.io/klog/v2/ktesting"
+	_ "k8s.io/klog/v2/ktesting/init"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/feature"
+	"k8s.io/kubernetes/pkg/scheduler/framework/runtime"
+)
+
+var (
+	podName       = "my-pod"
+	podUID        = types.UID("1234")
+	resourceName  = "my-resource"
+	resourceName2 = resourceName + "-2"
+	claimName     = podName + "-" + resourceName
+	claimName2    = podName + "-" + resourceName + "-2"
+	className     = "my-resource-class"
+	namespace     = "default"
+	templateName  = "my-template"
+
+	resourceClass = toResourceClass(resourcev1alpha1ac.ResourceClass(className))
+
+	podWithClaimName      = toPod(corev1ac.Pod(podName, namespace).WithUID(podUID).WithSpec(corev1ac.PodSpec().WithResourceClaims(corev1ac.PodResourceClaim().WithName(resourceName).WithSource(corev1ac.ClaimSource().WithResourceClaimName(claimName)))))
+	otherPodWithClaimName = toPod(corev1ac.Pod(podName, namespace).WithUID(podUID + "-II").WithSpec(corev1ac.PodSpec().WithResourceClaims(corev1ac.PodResourceClaim().WithName(resourceName).WithSource(corev1ac.ClaimSource().WithResourceClaimName(claimName)))))
+	podWithClaimTemplate  = toPod(corev1ac.Pod(podName, namespace).WithUID(podUID).WithSpec(corev1ac.PodSpec().WithResourceClaims(corev1ac.PodResourceClaim().WithName(resourceName).WithSource(corev1ac.ClaimSource().WithResourceClaimTemplateName(templateName)))))
+	podWithTwoClaimNames  = toPod(corev1ac.Pod(podName, namespace).WithUID(podUID).WithSpec(corev1ac.PodSpec().WithResourceClaims(corev1ac.PodResourceClaim().WithName(resourceName).WithSource(corev1ac.ClaimSource().WithResourceClaimName(claimName)), corev1ac.PodResourceClaim().WithName(resourceName2).WithSource(corev1ac.ClaimSource().WithResourceClaimName(claimName2)))))
+
+	workerNode = toNode(corev1ac.Node("worker").WithLabels(map[string]string{"nodename": "worker"}))
+
+	pendingImmediateClaim           = toResourceClaim(resourcev1alpha1ac.ResourceClaim(claimName, namespace).WithSpec(resourcev1alpha1ac.ResourceClaimSpec().WithAllocationMode(resourcev1alpha1.AllocationModeImmediate).WithResourceClassName(className)))
```

Comment:
Prefer this, for readability:
```golang
  a.WithFoo().
    WithBar().
```

Reply:
Agreed, formatting as a single line isn't good. I'll update that.